summary/summary.py

```diff
@@ -69,6 +69,22 @@ def content_object_init(instance):
     instance._summary = summary
     instance.has_summary = True
 
+
+def run_plugin(generators):
+    for generator in generators:
+        if isinstance(generator, ArticlesGenerator):
+            for article in generator.articles:
+                extract_summary(article)
+        elif isinstance(generator, StaticGenerator):
+            for article in generator.staticfiles:
+                extract_summary(article)
```

Comment:
Err... `StaticGenerator`? `Static` objects don't have any `content` to extract a `summary`.